instrumentation/opentelemetry-instrumentation-sqlalchemy/src/opentelemetry/instrumentation/sqlalchemy/engine.py

```diff
@@ -77,6 +77,17 @@ def _wrap_create_engine_internal(func, module, args, kwargs):
     return _wrap_create_engine_internal
 
 
+def _wrap_connect(tracer_provider=None):
+    def _wrap_connect_internal(func, module, args, kwargs):
+        tracer = trace.get_tracer(
+            module.name,
+            __version__,
+            tracer_provider=tracer_provider,
+        )
```

Comment:
Shouldn't be creating a new tracer everytime

Reply:
How does it create a new tracer? 

Is it different then the other wrappers (e.g. `_wrap_create_engine`) which use `_get_tracer` (which in turn use `trace.get_tracer` too)? Because that's where I took the inspiration from.

Reply:
>Is it different then the other wrappers (e.g. _wrap_create_engine)

People don't really create engines all the time. But on the other hand the `connect` is very commonly used and it's unnecessary to create tracer on each `connect`. Create a tracer at the top of module and use it or pass it your wrapper function if it's already created a the wrap call point.

Reply:
Thanks.

I've moved the tracer creation outside of the "connect" wrapped function.